### PR TITLE
Send reference request #1

### PR DESCRIPTION
**Pull request by stevehook**
## Context

Candidates will be able to send requests to referees as soon as they have filled in the details for that referee (or later from the review page).

## Changes proposed in this pull request

- Add `RequestController` and corresponding form to create new requests - it needs to send an email, progress the status and redirect to the review page.
- Add `CandidateNameController` and corresponding form to fill in missing names as required - this is triggered by the `RequestController`.
- System spec that covers the journey from the _Are you ready to send a reference request to ...?_ question through to the review page.
- `ReviewController` includes a `new` action that just renders the _Are you ready to send a reference request to ...?_ question to provide an entry point into this part of the flow. This will need to be integrated with the appropriate view/action maybe after https://github.com/DFE-Digital/apply-for-teacher-training/pull/3044 is merged.

To be covered in follow PR(s):
- Integration with the first part of the new reference journey including fixing the back button.
- Implement the links on the review page to send unsent reference requests.
- Automatic reference approval in Sandbox mode for `refbot1/2@example.com` - or is this handled elsewhere?

First screenshot is a temporary page that will need to be integrated to the rest of the journey:

image

image

image

## Guidance to review

- Are the controllers/actions/forms structured logically?
- Are the tests sufficient?
- Are there any issues you can foresee integrating with the rest of the new reference journey?
- To test on a review app or locally I think you will need to create a new reference, then navigate to `candidate/application/references/request/:id` to get started

## Link to Trello card

https://trello.com/c/50ncVpYS/2215-💔-dev-send-reference-request-mvp

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Review by paulrobertlloyd — APPROVED** [review]
LGTM. Will test when other PRs are merged and can test end-to-end.

File: spec/forms/candidate_interface/reference/candidate_name_form_spec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+require 'rails_helper'
+
+RSpec.describe CandidateInterface::Reference::CandidateNameForm, type: :model do
+  describe 'validations' do
+    it { is_expected.to validate_presence_of(:first_name) }
+    it { is_expected.to validate_presence_of(:last_name) }
+  end
+
+  describe '.build_from_reference' do
+    it 'creates an object based on the reference' do
```
Comment (malcolmbaig) [inline spec/forms/candidate_interface/reference/candidate_name_form_spec.rb:10]: `populates the form based on the reference`?

File: spec/system/candidate_interface/decoupled_references/candidate_requests_a_references_spec.rb [hunk 2]
```diff
@@ -0,0 +1,118 @@
+require 'rails_helper'
```
Comment (malcolmbaig) [inline spec/system/candidate_interface/decoupled_references/candidate_requests_a_references_spec.rb:1]: typo in the filename s/references/reference/

**Review by malcolmbaig — APPROVED** [review]
lgtm, couple of minor comments

**Pull request merged**